src/BuiltInTools/dotnet-watch/Program.cs

```diff
@@ -129,52 +131,62 @@ internal static RootCommand CreateRootCommand(Func<CommandLineOptions, Task<int>
                 return null;
             });
 
-            var projectOption = new Option<string>("-p", "The project to watch") { IsHidden = true };
+            var listOption = new Option<bool>(
+                    "--list",
+                    "Lists all discovered files without starting the watcher");
+
+            var shortProjectOption = new Option<string>("-p", "The project to watch") { IsHidden = true };
+            var longProjectOption = new Option<string>("--project","The project to watch");
+            var noHotReloadOption = new Option<bool>(
+                    new[] { "--no-hot-reload" },
+                    "Suppress hot reload for supported apps.");
             var root = new RootCommand(Description)
             {
                  quiet,
                  verbose,
-                 new Option<bool>(
-                    new[] { "--no-hot-reload" },
-                    "Suppress hot reload for supported apps."),
-                 new Option<string>(
-                     "--project",
-                    "The project to watch"),
-                 projectOption,
-                 new Option<bool>(
-                    "--list",
-                    "Lists all discovered files without starting the watcher"),
+                 noHotReloadOption,
+                 longProjectOption,
+                 shortProjectOption,
+                 listOption
             };
-
+            
             root.TreatUnmatchedTokensAsErrors = false;
-            root.Handler = CommandHandler.Create((CommandLineOptions options, ParseResult parseResults) =>
-            {
-                if (string.IsNullOrEmpty(options.Project))
+            root.SetHandler((ParseResult parseResults) => {
+                var projectValue = parseResults.GetValueForOption(longProjectOption);
```

Comment:
nit: it would be good to extract to method probably

Reply:
With your comment above it can be extracted to the binder, it's not necessary for the actual handler logic.